### PR TITLE
Merge final result from DRSS 2022

### PR DESCRIPTION
**Pull request by rohansaw**
- fix sizing problem by using haversine formula and decimals
- added custom project setings for generation
- allow to add actual side distance of signals instead of a hard-coded number
- further minor adaptions to be compatible with geo_nodes
- further minor changes

### CODE REVIEW

**Comment by arneboockmeyer** [comment]
@rohansaw Could you please merge the main branch into your branch to solve the conflicts?

**Comment by rohansaw** [comment]
> @rohansaw Could you please merge the main branch into your branch to solve the conflicts?

I merged it on my fork, however, I did not manage to test it yet and will not be able to do so the next days. So before a merge this should please be tested by someone.

File: generatorcli.py [hunk 1]
```diff
@@ -135,6 +135,11 @@ def find_edge_by_nodes(_node_a, _node_b):
             input_file_output.write(f"{command}\n")
     if command == "generate":
         generator = Generator()
-        generator.generate(nodes, edges, signals, filename)
+        config = Config(
+            author_name="Arne Boockmeier",
```
Comment (arneboockmeyer) [inline generatorcli.py:139]: This should be either set by some command or should be asked during the CLI process. In the end, its either `Boockmeyer` or some `Default Name`

File: planprogenerator/model/edge.py [hunk 2]
```diff
@@ -36,8 +36,12 @@ def get_length(self):
         return total_length
 
     def get_distance_between_two_geo_nodes(self, geo_node_a, geo_node_b):
-        min_x = min(geo_node_a.x, geo_node_b.x)
-        min_y = min(geo_node_a.y, geo_node_b.y)
-        max_x = max(geo_node_a.x, geo_node_b.x)
-        max_y = max(geo_node_a.y, geo_node_b.y)
-        return math.sqrt(math.pow(max_x - min_x, 2) + math.pow(max_y - min_y, 2))
+        pi_over_180  = Decimal(math.pi/180)
```
Comment (arneboockmeyer) [inline planprogenerator/model/edge.py:39]: There needs to be a switch here

File: planprogenerator/model/node.py [hunk 3]
```diff
@@ -81,11 +82,15 @@ def calc_anschluss_of_all_nodes(self):
 
         def get_arc_between_nodes(_node_a, _node_b):
             def length_of_side(__node_a, __node_b):
-                __min_x = min(__node_a.geo_node.x, __node_b.geo_node.x)
-                __min_y = min(__node_a.geo_node.y, __node_b.geo_node.y)
-                __max_x = max(__node_a.geo_node.x, __node_b.geo_node.x)
-                __max_y = max(__node_a.geo_node.y, __node_b.geo_node.y)
-                return math.sqrt(math.pow(__max_x - __min_x, 2) + math.pow(__max_y - __min_y, 2))
+                pi_over_180  = Decimal(math.pi/180)
```
Comment (arneboockmeyer) [inline planprogenerator/model/node.py:85]: There need to be a switch here aswell

File: planprogenerator/utils/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+from .config import Config
```
Comment (arneboockmeyer) [inline planprogenerator/utils/__init__.py:1]: Line at end of file

File: planprogenerator/utils/helpers.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+from typing import Literal
+import pyproj
+
+from planprogenerator.model.node import Node
+
+def dbref_coords(x, y, coord_representation: Literal['wgs84', 'dbref']):
```
Comment (arneboockmeyer) [inline planprogenerator/utils/helpers.py:6]: Is this method converting the coordinates to DB Ref?

**Review by arneboockmeyer — CHANGES_REQUESTED** [review]
We could think about moving all the coords stuff (representation, helper methods) to the new railway model. We should also move the config there.

**Comment by jeriox** [comment]
replaced by #10

**Pull request closed without merging**